### PR TITLE
 Do not specifiy files if they're set to None #406

### PR DESCRIPTION
**Pull request by mandaltapesh**
Hi armenzg,

This is the PR which adds the functionality to notify the user if he/she passes an empty file as parameter. The PR also includes test for it.The errors pointed out by flake8 has been resolved as far as possible.

r? armenzg

Regards,
Tapesh Mandal
IRC nick: tapesh

### CODE REVIEW

File: buildapi_client/buildapi_client.py [hunk 1]
```diff
@@ -52,6 +56,9 @@ def trigger_arbitrary_job(repo_name, builder, revision, auth, files=[], dry_run=
     )
     if req.status_code == 401:
         raise BuildapiAuthError("Your credentials were invalid. Please try again.")
+    
+    if files == [None, None]:
```
Comment (armenzg) [inline buildapi_client/buildapi_client.py:60]: Could you please use any()?

Comment (armenzg) [inline buildapi_client/buildapi_client.py:60]: It helps in cases where more than 2 files are passed in.

File: test/test_buildapi_client.py [hunk 2]
```diff
@@ -66,6 +66,13 @@ def test_bad_response(self, post):
         with self.assertRaises(buildapi_client.BuildapiAuthError):
             buildapi_client.trigger_arbitrary_job(
                 "repo", "builder", "123456123456", auth=None, dry_run=False)
+    
+    @patch('requests.post', return_value=mock_response(POST_RESPONSE, [None, None]))
+    def test_files(self, post):
+        """trigger_arbitrary_job should raise an BuildapiError if it receives files set to [None, None]."""
+        with self.assertRaises(buildapi_client.buildapi_client.BuildapiError):
+            buildapi_client.trigger_arbitrary_job(
+                repo_name="repo", builder="builder", revision="123456123456", auth=None, files=[None, None], dry_run=False, extra_properties=None)
```
Comment (armenzg) [inline test/test_buildapi_client.py:75]: Also add one test for when files=[] and files=[None]

**Comment by armenzg** [comment]
Good job on your code changes.

We need to fix the flake8 issues and the other comments I raised.

**Comment by mandaltapesh** [comment]
hi armenzg,
i will open a new PR and have it ready by monday .  then you can close this and merge the new one

**Comment by mandaltapesh** [comment]
Hi armenzg,
i think we should drop the idea of any() because if nothing is passed in the files it returns false . What i observed was there were other tests written for other purposes which did not pass files parameter at all so it may interfere with it .
https://docs.python.org/2/library/functions.html#any

r? armenzg 

regards,
tapesh

File: buildapi_client/buildapi_client.py (hunk 1 above)
Comment (armenzg) [inline buildapi_client/buildapi_client.py:60]: Hi @mandaltapesh 
You can reply to my comments in context instead of at the bottom of the PR.

You can do this:

> if files is not None and any(files)

``` python
In [7]: a = None
In [6]: a is not None and any(a)
Out[6]: False
In [7]: a = []
In [8]: a is not None and any(a)
Out[8]: False
In [9]: a = [None]
In [10]: a is not None and any(a)
Out[10]: False
In [11]: a = [None, None]
In [12]: a is not None and any(a)
Out[12]: False
```

**Comment by armenzg** [comment]
I figured out why the test was failling.
It is perhaps because we're not using Mock() properly but we can work around it for now.

With @patch we fake the response of requests.post and we return a mock object [1]

Could you write the test like this?
https://github.com/armenzg/buildapi_client/blob/master/test/test_buildapi_client.py#L40

This was a a learning opportunity for me as well :+1: 

[1]

``` python
In [38]: a = mock_response(POST_RESPONSE, 200)

In [39]: b = mock_response(POST_RESPONSE, 200)

In [40]: a == b
Out[40]: False

In [41]: a
Out[41]: <Mock id='140269806971536'>
```
